scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/RMProxyActiveObject.java

```diff
@@ -179,8 +183,14 @@ public synchronized void cleanCallBack(Future<ScriptResult<?>> future) {
             Entry<Node, ScriptResult<?>> entry = iterator.next();
             if (!PAFuture.isAwaited(entry.getValue())) { // !awaited = arrived
                 if (logger.isInfoEnabled()) {
-                    logger.info("Cleaning script successfull, node freed : " +
-                        entry.getKey().getNodeInformation().getURL());
+                    String nodeUrl = entry.getKey().getNodeInformation().getURL();
+                    logger.info("Cleaning script successfull, node freed : " + nodeUrl
+                        );
+                    try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)